internal/pkg/vm/testing.go

```diff
@@ -222,25 +200,21 @@ type testStorage struct {
 }
 
 // NewTestStorage returns a new "testStorage"
-func NewTestStorage(state interface{}) runtime.Storage {
+func NewTestStorage(state interface{}) *testStorage {
 	return &testStorage{
 		state: state,
 	}
 }
 
-var _ runtime.Storage = (*testStorage)(nil)
+var _ specsruntime.Store = (*testStorage)(nil)
 
-func (ts *testStorage) Put(v interface{}) cid.Cid {
+func (ts *testStorage) Put(v specsruntime.CBORMarshaler) cid.Cid {
 	ts.state = v
 	if cm, ok := v.(cbg.CBORMarshaler); ok {
 		buf := new(bytes.Buffer)
 		err := cm.MarshalCBOR(buf)
 		if err == nil {
-			nd, err := cbor.Decode(buf.Bytes(), types.DefaultHashFunction, -1)
-			if err != nil {
-				panic("failed to decode")
-			}
-			return nd.Cid()
+			return cid.NewCidV1(cid.Raw, buf.Bytes())
```

Comment:
I realise this is a test storage, but using raw-type CIDs here makes me nervous. I suggest using the cid.V1Builder with DAG_CBOR, as for other CIDs throughout gfc.

Reply:
sure, that is just keeping whatever needed the old context working, this is all likely to get redone once we start adding the new vm tests.